assets/src/edit-story/components/checklist/checks/videoOptimization.js

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { _n, __ } from '@web-stories-wp/i18n';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useCallback, useMemo, useRef } from 'react';
+import { useFeature } from 'flagged';
+import { useConfig, useLocalMedia, useStory } from '../../../app';
+import { VIDEO_SIZE_THRESHOLD } from '../../../app/media/utils/useFFmpeg';
+import { PRIORITY_COPY } from '../constants';
+import { LayerThumbnail, Thumbnail, THUMBNAIL_TYPES } from '../../thumbnail';
+import {
+  CARD_TYPE,
+  ChecklistCard,
+  DefaultFooterText,
+} from '../../checklistCard';
+import { filterStoryElements } from '../utils/filterStoryElements';
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  Tooltip,
+} from '../../../../design-system';
+import { useHighlights } from '../../../app/highlights';
+import { useRegisterCheck } from '../checkCountContext';
+import { StyledVideoOptimizationIcon } from '../../checklistCard/styles';
+
+const STATES = {
+  UPLOADING: 'UPLOADING',
+  UPLOADED: 'UPLOADED',
+};
+
+const OptimizeButton = styled(Button)`
+  margin-top: 4px;
+`;
+
+export function isVideoElementOptimized(element = {}) {
+  if (element.resource?.isTranscoding) {
+    return true;
+  }
+
+  if (element.resource?.local || element.resource?.isOptimized) {
+    return false;
+  }
```

Comment:
The function is called `isVideoElementOptimized` but it returns `false` when `element.resource?.isOptimized`? 🤔

Reply:
`videoElementNotOptimized` would be a better name for this filter.